### ISSUE TITLE
Question about code in blog post

### ISSUE THREAD

**Issue by mcastorina** [domenicquirl/blog#2]
Hey! I was going through your blog post and actually implementing the code when I noticed the following snippet [here](https://domenicquirl.github.io/blog/parsing-basics/#statements).

```rust
#[derive(Debug, Clone, PartialEq)]
pub enum Stmt {
    Let {
        var_name: String,
        value:    Box<Expr>,      // <-------
    },
    Assignment {
        var_name: String,
        value:    Box<Expr>,      // <-------
    },
    IfStmt {
        condition: Box<Expr>,     // <-------
        body:      Vec<Stmt>,
        else_stmt: Option<Box<Stmt>>,
    },
    Block {
        stmts: Vec<Stmt>,
    },
}
```

I think these `Box`es around `Expr` aren't needed, but I'm not sure. Is there a reason for it or is it another copy paste error? :)

In any case, the code still works with the `Box`, so I'm more curious if there is an intention behind using it. Thanks!

**Comment by domenicquirl** [comment]
It is definitely not an "error". As you note, it works either way, and I just chose to follow the same pattern here as in `Expr` itself, where you do need the indirection.

There are only so many concepts a brain can handle from a single source of information, and since I wrote a 3000 line post I feel like I'm already stretching that budget fairly thin. So mostly I tried to not have to talk about the implementation of the syntax tree (as opposed to the concept of having one, which is discussed). 

The AST is already not particularly efficient, and performance is in no way the point of the article, so I don't think it matters much.
